src/endpoints/users_groups/impl.cpp

```diff
@@ -131,805 +141,958 @@ namespace
+    IRODS_HTTP_API_HANDLER_FUNCTION_SIGNATURE(handle_remove_from_group_op)
     {
-        const auto result = irods::http::resolve_client_identity(_req);
+        auto result = irods::http::resolve_client_identity(_req);
         if (result.response) {
-            return *result.response;
+            return _sess_ptr->send(std::move(*result.response));
         }
 
         const auto* client_info = result.client_info;
-        log::info("{}: client_info = ({}, {})", __func__, client_info->username, client_info->password);
 
-        http::response<http::string_body> res{http::status::ok, _req.version()};
-        res.set(http::field::server, BOOST_BEAST_VERSION_STRING);
-        res.set(http::field::content_type, "application/json");
-        res.keep_alive(_req.keep_alive());
+        irods::http::globals::background_task([fn = __func__, client_info, _sess_ptr, _req = std::move(_req), _args = std::move(_args)] {
+            log::info("{}: client_info->username = [{}]", fn, client_info->username);
 
-        try {
-            const auto user_iter = _args.find("user");
-            if (user_iter == std::end(_args)) {
-                log::error("{}: Missing [user] parameter.", __func__);
-                return irods::http::fail(res, http::status::bad_request);
-            }
+            http::response<http::string_body> res{http::status::ok, _req.version()};
+            res.set(http::field::server, irods::http::version::server_name);
+            res.set(http::field::content_type, "application/json");
+            res.keep_alive(_req.keep_alive());
 
-            const auto group_iter = _args.find("group");
-            if (group_iter == std::end(_args)) {
-                log::error("{}: Missing [group] parameter.", __func__);
-                return irods::http::fail(res, http::status::bad_request);
-            }
+            try {
+                const auto user_iter = _args.find("user");
+                if (user_iter == std::end(_args)) {
+                    log::error("{}: Missing [user] parameter.", fn);
+                    return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+                }
 
-            auto conn = irods::get_connection(client_info->username);
+                const auto group_iter = _args.find("group");
+                if (group_iter == std::end(_args)) {
+                    log::error("{}: Missing [group] parameter.", fn);
+                    return _sess_ptr->send(irods::http::fail(res, http::status::bad_request));
+                }
+
+                auto conn = irods::get_connection(client_info->username);
 
-            const auto zone_iter = _args.find("zone");
-            if (zone_iter != std::end(_args)) {
-                adm::client::remove_user_from_group(conn, adm::group{group_iter->second}, adm::user{user_iter->second, zone_iter->second});
+                // TODO Should the zone parameter be required?
+                // I think it has to be to avoid abiguity.
```

Comment:
yes, let's make it required.   no magic.

Reply:
Done